cmd/agent.go

```diff
@@ -699,13 +699,73 @@ Examples:
 	},
 }
 
+var agentTestCmd = &cobra.Command{
+	Use:   "test",
+	Short: "Test an agent",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := env.NewLogger(cmd)
+
+		ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+		defer cancel()
+		theproject := project.EnsureProject(ctx, cmd)
+
+		agentID, _ := cmd.Flags().GetString("agent-id")
+		payload, _ := cmd.Flags().GetString("payload")
+
+		if agentID == "" {
+			keys, state := reconcileAgentList(logger, cmd, theproject.APIURL, theproject.Token, theproject)
+			if len(keys) == 0 {
+				tui.ShowWarning("no Agents found")
+				tui.ShowBanner("Create a new Agent", tui.Text("Use the ")+tui.Command("agent new")+tui.Text(" command to create a new Agent"), false)
+				return
+			}
+			var options []tui.Option
+			for _, v := range keys {
+				options = append(options, tui.Option{
+					ID:   state[v].Agent.ID,
+					Text: tui.PadRight(state[v].Agent.Name, 20, " ") + tui.Muted(state[v].Agent.ID),
+				})
+			}
+			agentID = tui.Select(logger, "Select an agent", "Select the agent you want to test", options)
+			agentID = strings.Replace(agentID, "agent_", "", 1)
+		}
+
+		if payload == "" {
+			payload = tui.Input(logger, "Enter the payload to send to the agent", "{\"hello\": \"world\"}")
+		}
+
+		apikey, err := agent.GetApiKey(context.Background(), logger, theproject.APIURL, theproject.Token, agentID)
+		if err != nil {
+			errsystem.New(errsystem.ErrApiRequest, err, errsystem.WithContextMessage("Failed to get agent API key")).ShowErrorAndExit()
+		}
+
+		client := util.NewAPIClient(ctx, logger, theproject.TransportURL, apikey)
+		var resp agent.Response[any]
+		if err := client.Do("POST", fmt.Sprintf("/webhook/%s", agentID), strings.NewReader(payload), &resp); err != nil {
```

Comment:
i think we should use the api endpoint since it will return.

we also need to allow content-type to be passed in (maybe assume application/octect-stream if not provided)

use tui.Success and error.

not sure we can use the API client though since it's really built for our API vs the generic HTTP client. Might want to just use the default http client.

probably should just stream out what you get back to os.Stdout instead of pulling it out.

Reply:
we probably need a --local true|false which allows you to switch between hitting your local agent vs the remote agent too

Reply:
I originally used the standard http package, but then realized we had a lot of code already for http requests in our client package. I'll do some testing with non-json payloads and probably go back to using the http package, since this is supposed to do a simple http request after all